pkgs/by-name/kb/kbd/package.nix

```diff
@@ -48,62 +58,88 @@ stdenv.mkDerivation rec {
     ./search-paths.patch
   ];
 
-  postPatch = ''
-    # Renaming keymaps with name clashes, because loadkeys just picks
-    # the first keymap it sees. The clashing names lead to e.g.
-    # "loadkeys no" defaulting to a norwegian dvorak map instead of
-    # the much more common qwerty one.
-    pushd data/keymaps/i386
-    mv qwertz/cz{,-qwertz}.map
-    mv olpc/es{,-olpc}.map
-    mv olpc/pt{,-olpc}.map
-    mv fgGIod/trf{,-fgGIod}.map
-    mv colemak/{en-latin9,colemak}.map
-    popd
-
-    # Fix paths to decompressors. Trailing space to avoid replacing `xz` in `".xz"`.
-    substituteInPlace src/libkbdfile/kbdfile.c \
-      --replace-fail 'gzip '  '${gzip}/bin/gzip ' \
-      --replace-fail 'bzip2 ' '${bzip2.bin}/bin/bzip2 ' \
-      --replace-fail 'xz '    '${xz.bin}/bin/xz ' \
-      --replace-fail 'zstd '  '${zstd.bin}/bin/zstd '
-
-    sed -i '
-      1i prefix:=$(vlock)
-      1i bindir := $(vlock)/bin' \
-      src/vlock/Makefile.in \
-      src/vlock/Makefile.am
-  '';
+  postPatch =
+    ''
+      # Renaming keymaps with name clashes, because loadkeys just picks
+      # the first keymap it sees. The clashing names lead to e.g.
+      # "loadkeys no" defaulting to a norwegian dvorak map instead of
+      # the much more common qwerty one.
+      pushd data/keymaps/i386
+      mv qwertz/cz{,-qwertz}.map
+      mv olpc/es{,-olpc}.map
+      mv olpc/pt{,-olpc}.map
+      mv fgGIod/trf{,-fgGIod}.map
+      mv colemak/{en-latin9,colemak}.map
+      popd
+
+    ''
+    + lib.optionalString withCompression ''
+      # Fix paths to decompressors. Trailing space to avoid replacing `xz` in `".xz"`.
+      substituteInPlace src/libkbdfile/kbdfile.c \
+        --replace-fail 'gzip '  '${gzip}/bin/gzip ' \
+        --replace-fail 'bzip2 ' '${bzip2.bin}/bin/bzip2 ' \
+        --replace-fail 'xz '    '${xz.bin}/bin/xz ' \
+        --replace-fail 'zstd '  '${zstd.bin}/bin/zstd '
+    ''
+    + ''
+
+      sed -i '
+        1i prefix:=$(vlock)
+        1i bindir := $(vlock)/bin' \
+        src/vlock/Makefile.in \
+        src/vlock/Makefile.am
+    ''
+    + lib.optionalString (!withObsolete) ''
+      substituteInPlace src/Makefile.am \
+        --replace-fail 'dist_bin_SCRIPTS = unicode_start unicode_stop' 'dist_bin_SCRIPTS ='
+    '';
 
-  postInstall = ''
+  postInstall = lib.optionalString withObsolete ''
     for i in $out/bin/unicode_{start,stop}; do
       substituteInPlace "$i" \
         --replace /usr/bin/tty ${coreutils}/bin/tty
     done
   '';
 
-  buildInputs = [
-    check
-    pam
+  buildInputs =
+    [
+      check
+      pam
+    ]
+    ++ lib.optionals (withObsolete) [
+      # We install shell scripts as well. Due to strictDeps we must be
+      # explicit with the buildInputs.
+      bash
+    ];
+
+  disallowedReferences = lib.optionals (!withObsolete && !withCompression) [
```

Comment:
This deserves an explanation why !withCompression mean bash is is disallowed.

Reply:
I added an explanation. Good point.